static/app/views/starfish/utils/getDateFilters.tsx

```diff
@@ -5,12 +5,10 @@ import {PageFilters} from 'sentry/types';
 export const PERIOD_REGEX = /^(\d+)([h,d])$/;
 export const DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss';
 
-export function getDateFilters(selection: PageFilters) {
-  const [_, num, unit] = selection.datetime.period?.match(PERIOD_REGEX) ?? [];
+export function getDateFilters(selection: Partial<PageFilters['datetime']>) {
```

Comment:
This util is specific to the date filter, so rather than accepting the entire page filter state (`PageFilters`) as input, it should just accept the `datetime` part (`PageFilters['datetime']`).

Reply:
Should this be a Partial?